pinot-segment-local/src/main/java/org/apache/pinot/segment/local/upsert/merger/PartialUpsertColumnarMerger.java

```diff
@@ -77,11 +77,7 @@ public void merge(LazyRow previousRow, GenericRow newRow, Map<String, Object> re
       if (!(merger instanceof OverwriteMerger)) {
         Object prevValue = previousRow.getValue(column);
         if (prevValue != null) {
-          if (newRow.isNullValue(column)) {
-            resultHolder.put(column, prevValue);
-          } else {
-            resultHolder.put(column, merger.merge(prevValue, newRow.getValue(column)));
-          }
+          resultHolder.put(column, merger.merge(prevValue, newRow.getValue(column)));
```

Comment:
This change is backward-incompatible. Going through the attached issue, I would recommend to create a new merger strategy and allow null-overwrites there.

Reply:
Okay 👍🏻

Reply:
Done!